### ISSUE TITLE
Incorrect velocity calculation

### ISSUE THREAD

**Issue by TJStienstra** [sympy/sympy#23358]
As already pointed out in #22956 an incorrect assumption seems to be added in [21730](https://github.com/sympy/sympy/pull/21730).

Here is an example:
```py
import sympy as sm
import sympy.physics.mechanics as me

q1, q2 = me.dynamicsymbols('q1:3')
u1, u2 = sm.symbols('u1:3')
A = me.ReferenceFrame('A')
B = me.ReferenceFrame('B')
C = me.ReferenceFrame('C')
B.orient_axis(A, A.z, q1)
B.set_ang_vel(A, u1*A.z)
C.orient_axis(B, B.z, q2)
C.set_ang_vel(B, u2*B.z)

Am = me.Point('Am')
Am.set_vel(A, 0)
Bm = me.Point('Bm')
Bm.set_pos(Am, B.x)
Bm.set_vel(B, 0)
Bm.set_vel(C, 0)
Cm = me.Point('Cm')
Cm.set_pos(Bm, C.x)
Cm.set_vel(C, 0)
temp = Cm._vel_dict.copy()
print(Cm.vel(A))  # Takes the position derivative
Cm._vel_dict = temp
Cm.v2pt_theory(Bm, B, C)
print(Cm.vel(A))  # Uses the velocity from an intermediate frame
```
This outputs:
```py
u1*B.y + (u1 + u2)*C.y
u2*C.y
```
Which should be the same, but as shown it isn't. In fact with the code implemented in [21730](https://github.com/sympy/sympy/pull/21730) the incorrect assumption is made that:
<img src="https://render.githubusercontent.com/render/math?math={{}^Av^{Cm} = {}^Nv^{Am} %2B {}^Bv^{Cm}}#gh-light-mode-only">
<img src="https://render.githubusercontent.com/render/math?math={\color{white}{}^Av^{Cm} = {}^Nv^{Am} %2B {}^Bv^{Cm}}#gh-dark-mode-only">
Because one the of components of the position vector from `Am` to `Cm` is expressed in the `B` frame.

@sidhu1012 or @moorepants could you elaborate why this assumption could normally be made? Also the test that was added, namely [`test_vel_frame`](https://github.com/sympy/sympy/blob/4c969e4ded0ab76a1a22fa3c93bc8f22af854e22/sympy/physics/vector/tests/test_point.py#L302) is also not really correct seen from a mathematical/mechanical point of view. Since the `N` frame is equal to the `A` frame the distance between `P` and `P1` is constant, so the velocity of `P` should be the same as `P1`.

PS As always correct me if I'm wrong, I am still learning more about mechanics.

**Comment by moorepants** [comment]
> PS As always correct me if I'm wrong, I am still learning more about mechanics.

I think you are probably right.

**Comment by moorepants** [comment]
Yes, you are right. The `vel + neighbor_velocity` is not valid. We shouldn't have merged #21730.
